mitiq/folding_cirq.py

```diff
@@ -1,76 +1,197 @@
-from typing import List, Callable
-import random
-from cirq import Circuit, inverse
+"""Functions to fold gates in Cirq circuits."""
+from copy import deepcopy
+from typing import (Callable, Iterable, List)
+
+import numpy as np
+
+from cirq import (Circuit, InsertStrategy, inverse)
+
+
+MAX_STRETCH_FACTOR = 100
```

Comment:
Can you say more about why you included this @rmlarose ?

Reply:
Mostly because of runtime considerations, especially with making deepcopy's every iteration. Generally I think it's not useful to stretch past a certain point, though, so we might want to consider some cutoff. The number `100` is pretty arbitrary.

Reply:
Given that the limitation seems arbitrary, I recommend that it be removed for now.